app/react/kubernetes/cluster/ConfigureView/ConfigureForm/useStorageClasses.ts

```diff
@@ -26,9 +26,31 @@ export const availableStorageClassPolicies = [
   },
 ];
 
-export function useStorageClassesFormValues(
-  environment: Environment | null | undefined
-) {
+export function useStorageClasses(environment?: Environment | null) {
+  return useQuery(
+    [
+      'environments',
+      environment?.Id,
+      'kubernetes',
+      'storageclasses',
+      // include the storage classes in the cache key to force a refresh when the storage classes change in the environment object
+      JSON.stringify(environment?.Kubernetes.Configuration.StorageClasses),
```

Comment:
What if Kubernetes/Kubernetes.Configutration/`Kubernetes.Configutration.StorageClasses` is undefined or null?

Reply:
It can be undefined, which will give undefined in the last query key, which is still allowed